core/Object.php

```diff
@@ -816,7 +816,9 @@ protected function addMethodsFrom($property, $index = null) {
 		}
 
 		if(method_exists($extension, 'allMethodNames')) {
+			if (is_a($extension, 'Extension')) $extension->setOwner($this);
```

Comment:
I think we prefer `instanceof Extension`, since is_a has some php-version sensitive behaviour.